algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py

```diff
@@ -0,0 +1,182 @@
+    if isinstance(value, dict):
+      param_types_dict[name] = _param_types(value)
+    else:
+      if 'bias' in name:
+        param_types_dict[name] = spec.ParameterType.BIAS
+      else:
+        param_types_dict[name] = spec.ParameterType.WEIGHT
+  return param_types_dict
+
+
+class CifarWorkload(BaseCifarWorkload):
+
+  def _build_dataset(self,
+                     data_rng: spec.RandomState,
+                     split: str,
+                     data_dir: str,
+                     batch_size: int,
+                     cache: Optional[bool] = None,
+                     repeat_final_dataset: Optional[bool] = None,
+                     num_batches: Optional[int] = None):
+    if batch_size % jax.local_device_count() > 0:
+      raise ValueError('Batch size must be divisible by the number of devices')
+    ds_builder = tfds.builder('cifar10', data_dir=data_dir)
+    ds_builder.download_and_prepare()
+    train = split == 'train'
+    ds = input_pipeline.create_input_iter(
+        split,
+        ds_builder,
+        data_rng,
+        batch_size,
+        self.train_mean,
+        self.train_stddev,
+        self.center_crop_size,
+        self.resize_size,
+        self.aspect_ratio_range,
+        self.scale_ratio_range,
+        train=train,
+        cache=not train if cache is None else cache,
+        repeat_final_dataset=repeat_final_dataset,
+        num_batches=num_batches)
+    return ds
+
+  def sync_batch_stats(self, model_state):
+    """Sync the batch statistics across replicas."""
+    # An axis_name is passed to pmap which can then be used by pmean.
+    # In this case each device has its own version of the batch statistics and
+    # we average them.
+    avg_fn = jax.pmap(lambda x: lax.pmean(x, 'x'), 'x')
+    new_model_state = model_state.copy(
+        {'batch_stats': avg_fn(model_state['batch_stats'])})
+    return new_model_state
+
+  @property
+  def param_shapes(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    return self._param_shapes
+
+  @property
+  def model_params_types(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    if self._param_types is None:
+      self._param_types = _param_types(self._param_shapes.unfreeze())
+    return self._param_types
+
+  # Return whether or not a key in spec.ParameterContainer is the output layer
+  # parameters.
+  def is_output_params(self, param_key: spec.ParameterKey) -> bool:
+    pass
+
+  def initialized(self, key, model):
+    input_shape = (1, 32, 32, 3)
+    variables = jax.jit(model.init)({'params': key},
+                                    jnp.ones(input_shape, model.dtype))
+    model_state, params = variables.pop('params')
+    return params, model_state
```

Comment:
nit: can we inline this in `init_model_fn` just so that people aren't tempted to call it?

Reply:
Done!